train/create_predictions.py

```diff
@@ -99,25 +140,26 @@ def get_suggestions(detector, basedir: str, untagged_output: str,
                 all_tagged.extend(list(reader))
         already_tagged = defaultdict(set)
         for row in all_tagged:
-            already_tagged[row[FOLDER_LOCATION]].add(row[0])
+            already_tagged[row[FOLDER_LOCATION]].add(row[FILENAME_LOCATION])
+        print("Already tagged {0} images.".format(sum(map(len, already_tagged.values()))))
         subdirs = [subfile for subfile in basedir.iterdir() if subfile.is_dir()]
         print("subdirs: ", subdirs)
-        all_names = []
-        all_image_files = [] 
         all_sizes = []
         for subdir in subdirs:
-            cur_image_names = list(subdir.rglob(filetype))
-            print("Total image names: ", len(cur_image_names))
-            all_image_files += [str(image_name) for image_name in cur_image_names]
             foldername = subdir.stem
-            all_names += [(foldername, filename.name) for filename in cur_image_names]
-        # Reversed because numpy is row-major
-        all_sizes = [cv2.imread(image, CV2_COLOR_LOAD_FLAG).shape[:2] for image in all_image_files]
-        all_images = np.zeros((len(all_image_files), *reversed(image_size), NUM_CHANNELS), dtype=np.uint8)
-        for curindex, image in enumerate(all_image_files):
-            all_images[curindex] = cv2.resize(cv2.imread(image, CV2_COLOR_LOAD_FLAG), image_size)
+            already_tagged_this_folder = already_tagged[foldername]
+            all_images_this_folder, all_names_this_folder, all_sizes_this_folder =\
+                get_images_for_prediction(subdir, filetype, already_tagged_this_folder, image_size)
+
+            all_names += all_names_this_folder
+            if (all_images == None):
+                all_images = all_images_this_folder
+            else:
+                all_images = np.concatenate((all_images,all_images_this_folder), axis=0)
```

Comment:
I would suggest either getting rid of the np.concatenate or doing it at the end instead of at each folder.